cmd/rpcdaemon/commands/engine_api.go

```diff
@@ -19,32 +19,38 @@ import (
 // ExecutionPayload represents an execution payload (aka slot/block)
 type ExecutionPayload struct {
 	ParentHash    common.Hash     `json:"parentHash"    gencodec:"required"`
+	FeeRecipient  common.Address  `json:"feeRecipient"  gencodec:"required"`
 	StateRoot     common.Hash     `json:"stateRoot"     gencodec:"required"`
-	ReceiptRoot   common.Hash     `json:"receiptRoot"   gencodec:"required"`
+	ReceiptsRoot  common.Hash     `json:"receiptsRoot"  gencodec:"required"`
 	LogsBloom     hexutil.Bytes   `json:"logsBloom"     gencodec:"required"`
 	Random        common.Hash     `json:"random"        gencodec:"required"`
 	BlockNumber   hexutil.Uint64  `json:"blockNumber"   gencodec:"required"`
 	GasLimit      hexutil.Uint64  `json:"gasLimit"      gencodec:"required"`
 	GasUsed       hexutil.Uint64  `json:"gasUsed"       gencodec:"required"`
-	Extra         hexutil.Bytes   `json:"extraData"     gencodec:"required"`
+	Timestamp     hexutil.Uint64  `json:"timestamp"     gencodec:"required"`
+	ExtraData     hexutil.Bytes   `json:"extraData"     gencodec:"required"`
 	BaseFeePerGas *hexutil.Big    `json:"baseFeePerGas" gencodec:"required"`
 	BlockHash     common.Hash     `json:"blockHash"     gencodec:"required"`
-	Timestamp     hexutil.Uint64  `json:"timestamp"     gencodec:"required"`
-	Coinbase      common.Address  `json:"coinbase"      gencodec:"required"`
 	Transactions  []hexutil.Bytes `json:"transactions"  gencodec:"required"`
 }
 
-// PreparePayloadArgs represents a request to assemble a payload from the beacon chain
-type PreparePayloadArgs struct {
-	Random       *common.Hash    `json:"random"`
-	Timestamp    *hexutil.Uint64 `json:"timestamp"`
-	FeeRecipient *common.Address `json:"feeRecipient"`
+type ForkchoiceState struct {
```

Comment:
This is used for unwinds, not sure we need it since we are not going to do unwinds conventionally through fork choice but through executePayload

Reply:
Sounds unconventional, but OK, I've removed ForkchoiceState.